packages/ra-core/src/form/useSuggestions.ts

```diff
@@ -196,10 +194,22 @@ export const getSuggestionsFactory = ({
                     getChoiceValue(choice) === getChoiceValue(selectedItem)
             );
         } else {
-            suggestions = choices;
+            suggestions = [...choices];
         }
     } else {
-        suggestions = choices.filter(choice => matchSuggestion(filter, choice));
+        suggestions = choices.filter(
+            choice =>
+                matchSuggestion(filter, choice) ||
+                (selectedItem != null &&
```

Comment:
not sure why you added that, it doesn't seem related to react-final-form

Reply:
It was necessary for the choices inputs